tests/unit/test_connector.py

```diff
@@ -79,25 +70,46 @@ def test_connect_enable_iam_auth_error() -> None:
     instance = MockInstance(enable_iam_auth=False)
     mock_instances = {}
     mock_instances[connect_string] = instance
-    # set default connector
-    default_connector = connector.Connector()
-    connector._default_connector = default_connector
-    with patch.dict(default_connector._instances, mock_instances):
+    # init Connector
+    connector = Connector()
+    with patch.dict(connector._instances, mock_instances):
         # try to connect using enable_iam_auth=True, should raise error
         pytest.raises(
             ValueError,
-            default_connector.connect,
+            connector.connect,
             connect_string,
             "pg8000",
             enable_iam_auth=True,
         )
     # remove mock_instance to avoid destructor warnings
-    default_connector._instances = {}
+    connector._instances = {}
+
 
+def test_Connector_Init() -> None:
+    """Test that Connector __init__ sets default properties properly."""
+    connector = Connector()
+    assert connector._ip_type == IPTypes.PUBLIC
+    assert connector._enable_iam_auth is False
+    assert connector._timeout == 30
+    assert connector._credentials is None
+    connector.close()
 
-def test_default_Connector_Init() -> None:
-    """Test that default Connector __init__ sets properties properly."""
-    default_connector = connector.Connector()
+
+def test_global_connect() -> None:
+    """Test that global connect properly initializes a singleton Connector."""
+    connect_string = "test-project:test-region:test-instance"
+    # stub Connector.connect
+    setattr(Connector, "connect", stub_connect)
```

Comment:
Would it be better to stub the metadata and connect to the local mock server?

Reply:
This test is really meant to only test that calling the global connector initializes a `Connector` object with the proper defaults. Wouldn't testing that the global connector can make it all the way to connect to a server, be more of an integration test?

Reply:
The line between unit and integration test is only defined by what the unit(s) are. In this case, I think it's better to consistently stub/mock at the same place, so that we can ensure that parts are all working together. Since we're mocking the actual service I wouldn't really consider it an integration test.